packages/lesswrong/components/contents/HorizScrollBlock.tsx

```diff
@@ -130,11 +131,32 @@ const HorizScrollBlock = ({children, className, contentsClassName, classes}: {
   </div>
 }
 
-export default registerComponent(
-  'HorizScrollBlock',
-  HorizScrollBlock,
-  {styles, stylePriority: -1},
-);
-
-
-
+export const MaybeScrollableBlock = ({TagName, attribs, bodyRef, children}: {
+  TagName: any
```

Comment:
can we limit this to the list of known-valid tag names that it might actually be (i.e. `'p' | 'div' | 'table'`)?

Reply:
Attempting to narrow this type causes React to get confused about the type of the ref; it sees that `HTMLElement` is too broad, but can't map string tag names to corresponding element classes. So, no, we can't (without adding a cast-to-any somewhere else).